File: app/src/main/java/jp/gr/java_conf/foobar/testmaker/service/models/Quest.java
package jp.gr.java_conf.foobar.testmaker.service.models;

import io.realm.RealmList;
import io.realm.RealmObject;
import io.realm.annotations.PrimaryKey;
import io.realm.annotations.Required;

/**
 * Created by keita on 2017/02/08.
 */
public class Quest extends RealmObject {

    @PrimaryKey
    private long id;
    private String problem;
    private String answer;
    @Required
    private String explanation;
    private boolean correct;
    private String imagePath;
    private RealmList<Select> selections;
    private RealmList<Select> answers;
    private int type;
    private boolean auto;
    private boolean solving;

    public long getId() {
        return id;
    }

    public void setProblem(String p) {
        problem = p;
    }

    public String getProblem() {
        return problem;
    }

    public String getProblem(boolean isReverse){

        if(isReverse) return answer;

        return problem;

    }

    public void setAnswer(String a) {
        answer = a;
    }

    public String getAnswer() {
        return answer;
    }

    public String getAnswer(boolean isReverse){

        if(isReverse) return problem;

        return answer;

    }

    public String getExplanation() {
        return explanation;
    }

    public void setExplanation(String s) {
        explanation = s;
    }

    public void setImagePath(String i) {
        imagePath = i;
    }

    public String getImagePath() {
        return imagePath;
    }

    public void setType(int t) {
        type = t;
    }

    public int getType() {
        return type;
    }

    public void setAuto(boolean a) {
        auto = a;
    }

    public boolean getAuto() {
        return auto;
    }

    public void setCorrect(boolean c) {
        correct = c;
    }

    public boolean getCorrect() {
        return correct;
    }

    public void setSolving(boolean c) {
        solving = c;
    }

    public boolean getSolving() {
        return solving;
    }

    public void setSelections(String[] strings) {

        selections.clear();

        for (int i = 0; i < strings.length; i++) {
            Select select = new Select();
            select.setSelection(strings[i]);
            selections.add(select);
        }

    }

    public RealmList<Select> getSelections() {
        return selections;
    }

    public void setAnswers(String[] strings) {

        answers.clear();

        for (int i = 0; i < strings.length; i++) {
            Select select = new Select();
            select.setSelection(strings[i]);
            answers.add(select);
        }

    }

    public RealmList<Select> getAnswers() {
        return answers;
    }



}

File: app/src/main/java/jp/gr/java_conf/foobar/testmaker/service/models/Quest.kt
package jp.gr.java_conf.foobar.testmaker.service.models

import io.realm.RealmList
import io.realm.RealmObject
import io.realm.annotations.PrimaryKey
import io.realm.annotations.Required

/**
 * Created by keita on 2017/02/08.
 */
open class Quest : RealmObject() {

    @PrimaryKey
    var id: Long = 0
    var problem: String = ""
    var answer: String = ""
    @Required
    var explanation: String = ""
    var correct: Boolean = false
    var imagePath: String = ""
    var selections: RealmList<Select> = RealmList()
    var answers: RealmList<Select> = RealmList()
    var type: Int = 0
    var auto: Boolean = false
    var solving: Boolean = false
    var order: Int = 0
    var isCheckOrder: Boolean = false

    fun getProblem(isReverse: Boolean): String {

        return if (isReverse) answer else problem

    }

    fun getAnswer(isReverse: Boolean): String {

        return if (isReverse) problem else answer

    }

    fun setSelections(strings: Array<String>) {

        selections.clear()

        for (string in strings) {
            val select = Select()
            select.selection = string
            selections.add(select)
        }

    }

    fun setAnswers(strings: Array<String>) {

        answers.clear()

        for (string in strings) {
            val select = Select()
            select.selection = string
            answers.add(select)
        }

    }

}
